store/sql_reaction_store.go

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package store
+
+import (
+	"github.com/mattermost/platform/model"
+	"github.com/mattermost/platform/utils"
+
+	l4g "github.com/alecthomas/log4go"
+	"github.com/go-gorp/gorp"
+)
+
+const (
+	// set HasReactions = true iff the post has reactions, update UpdateAt only if HasReactions changes
+	UPDATE_POST_HAS_REACTIONS_QUERY = `UPDATE
+			Posts
+		SET
+			UpdateAt = (CASE
+				WHEN HasReactions != (SELECT count(0) > 0 FROM Reactions WHERE PostId = :PostId) THEN :UpdateAt
+				ELSE UpdateAt
+			END),
+			HasReactions = (SELECT count(0) > 0 FROM Reactions WHERE PostId = :PostId)
+		WHERE
+			Id = :PostId`
+)
```

Comment:
I'll move it down to where its used, but I think that's the best we can do now.

The only reason it's used in multiple places is because one calls `Exec` with a transaction and the other doesn't. If I run into this again, I'll reconsider wrapping both the `DbMap` and `Transaction` types in a single interface since they have identical `Exec` calls. That way, we won't have to repeat query strings even in cases like this.